src/extension/wps/wps-core/src/main/java/org/geoserver/wps/ppio/ProcessParameterIO.java

```diff
@@ -184,8 +184,12 @@ public int compare(ProcessParameterIO o1, ProcessParameterIO o2) {
         }
 
         // load by factory
-        List<PPIOFactory> ppioFactories = GeoServerExtensions
-                .extensions(PPIOFactory.class, context);
+        List<PPIOFactory> ppioFactories;
+        if (context != null) {
+            ppioFactories = GeoServerExtensions.extensions(PPIOFactory.class, context);
```

Comment:
Is it worth making this method more relaxed, ie if null context is passed in avoid producing a warning? 

What category of bugs is this warning catching to justify having two methods (and this kind of extra logic required to "Safely" call it?)